src/bootstrap/src/utils/exec.rs

```diff
@@ -49,6 +55,14 @@ impl OutputMode {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Default)]
+pub struct CommandCacheKey {
```

Comment:
We can just use the key directly in `BootstrapCommand`? To vaoid having to duplicate the fields.